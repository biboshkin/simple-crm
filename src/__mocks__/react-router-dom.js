jest.mock("react-router-dom", () => {
  // Require the original module to not be mocked...
  const originalModule = jest.requireActual("react-router-dom");

  return {
    __esModule: true,
    ...originalModule,
    // add your noops here
    useParams: jest.fn(),
    useHistory: jest.fn(),
  };
});
