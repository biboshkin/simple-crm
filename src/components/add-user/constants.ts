import { FieldProps, urls } from "../../constants";
import { phoneRule } from "../../utils";

const REQUIRED_MESSAGE = "Пожалуйста, заполните поле";
const EMAIL_MESSAGE = "Пожалуйста, введите корректный e-mail";
const PLACEHOLDER_DEFAULT = "Введите значение";

export const NAME = "ADD_USER";

export const TITLE = "Добавление пользователя";

export const SUCCESS_MESSAGE = "Пользователь успешно добавлен";

export const MANAGER_ROLE_ID = "733397ea-8436-44d4-9beb-2bc9e674e9e8";

export const FIELDS: FieldProps[] = [
  {
    fieldCode: "fullName",
    fieldName: "Ф.И.О.",
    fieldDescription: "",
    type: "string",
    readonly: false,
    disabled: false,
    span: { xl: 24, md: 24, lg: 24, sm: 24 },
    placeholder: PLACEHOLDER_DEFAULT,
    rules: [{ required: true, message: REQUIRED_MESSAGE }],
    permissions: [],
  },
  {
    fieldCode: "email",
    fieldName: "E-mail",
    fieldDescription: "",
    type: "email",
    readonly: false,
    disabled: false,
    span: { xl: 24, md: 24, lg: 24, sm: 24 },
    placeholder: PLACEHOLDER_DEFAULT,
    rules: [
      { required: true, message: REQUIRED_MESSAGE },
      {
        type: "email",
        message: EMAIL_MESSAGE,
      },
    ],
    permissions: [],
  },
  {
    fieldCode: "phone",
    fieldName: "Телефон",
    fieldDescription: "",
    type: "phone",
    readonly: false,
    disabled: false,
    span: { xl: 24, md: 24, lg: 24, sm: 24 },
    placeholder: "+7 ",
    rules: [phoneRule],
    permissions: [],
  },
  {
    fieldCode: "departmentId",
    fieldName: "Отдел",
    type: "entity",
    titleField: "departmentName",
    codeField: "id",
    readonly: false,
    disabled: false,
    rules: [{ required: true, message: REQUIRED_MESSAGE }],
    permissions: [],
    span: { xl: 24, md: 24, lg: 24, sm: 24 },
    _links: {
      self: {
        href: urls.departments.entity,
      },
    },
  },
];
