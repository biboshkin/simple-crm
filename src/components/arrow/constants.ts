export type ArrowDirection = "up" | "down";

export interface ArrowType {
  direction: ArrowDirection;
  isActive: boolean;
}
