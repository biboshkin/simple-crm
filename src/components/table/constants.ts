export const DEFAULT_PAGE_NUMBER = 1;
export const DEFAULT_PAGE_SIZE = 10;

export const FILTER_ALL_NAME = "all";

export interface SortColumnOrderProps {
  [key: string]: "ascend" | "descend";
}

export type SearchType = {
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onSearch: (value: string) => void;
  onClearAll: () => void;
};
