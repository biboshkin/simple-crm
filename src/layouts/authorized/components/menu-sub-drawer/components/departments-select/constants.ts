import { ReactNode } from "react";

export interface TreeDataProps {
  key: string;
  title: ReactNode;
  parentId?: string;
  children?: TreeDataProps[];
}
