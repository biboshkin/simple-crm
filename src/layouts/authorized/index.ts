export { default as AuthorizedLayout } from "./authorized";
