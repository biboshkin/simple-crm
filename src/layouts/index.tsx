export { UnauthorizedLayout } from "./unauthorized";
export { AuthorizedLayout } from "./authorized";
