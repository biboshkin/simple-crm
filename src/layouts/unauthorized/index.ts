export { default as UnauthorizedLayout } from "./unauthorized";
