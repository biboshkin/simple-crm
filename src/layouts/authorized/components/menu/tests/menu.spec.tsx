import React from "react";
import renderer from "react-test-renderer";

import { Menu } from "..";
import { shallow } from "enzyme";
import { Link, BrowserRouter } from "react-router-dom";
import { MENU_ITEMS } from "../../../../../constants/layouts";

test("render correct with collapsed false", () => {
  const component = renderer.create(
    <BrowserRouter>
      <Menu collapsed={false} />
    </BrowserRouter>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test("render correct with collapsed true", () => {
  const component = renderer.create(
    <BrowserRouter>
      <Menu collapsed={true} />
    </BrowserRouter>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test("correct Links count", () => {
  const wrapper = shallow(<Menu collapsed={false} />);
  expect(wrapper.find(Link).length).toBe(MENU_ITEMS.length);
});
