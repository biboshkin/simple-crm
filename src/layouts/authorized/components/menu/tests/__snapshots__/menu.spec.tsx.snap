// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correct with collapsed false 1`] = `
<ul
  className="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
  onMouseEnter={[Function]}
  onTransitionEnd={[Function]}
  role="menu"
  style={Object {}}
>
  <li
    className="ant-menu-item ant-menu-item-selected"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <span
      aria-label="user"
      className="anticon anticon-user"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="user"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
        />
      </svg>
    </span>
    <span>
      <a
        href="/clients"
        onClick={[Function]}
      >
        Клиенты
      </a>
    </span>
  </li>
  <li
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <span
      aria-label="calendar"
      className="anticon anticon-calendar"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="calendar"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
        />
      </svg>
    </span>
    <span>
      <a
        href="/tasks"
        onClick={[Function]}
      >
        Планы
      </a>
    </span>
  </li>
  <li
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <span
      aria-label="dollar"
      className="anticon anticon-dollar"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="dollar"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm47.7-395.2l-25.4-5.9V348.6c38 5.2 61.5 29 65.5 58.2.5 4 3.9 6.9 7.9 6.9h44.9c4.7 0 8.4-4.1 8-8.8-6.1-62.3-57.4-102.3-125.9-109.2V263c0-4.4-3.6-8-8-8h-28.1c-4.4 0-8 3.6-8 8v33c-70.8 6.9-126.2 46-126.2 119 0 67.6 49.8 100.2 102.1 112.7l24.7 6.3v142.7c-44.2-5.9-69-29.5-74.1-61.3-.6-3.8-4-6.6-7.9-6.6H363c-4.7 0-8.4 4-8 8.7 4.5 55 46.2 105.6 135.2 112.1V761c0 4.4 3.6 8 8 8h28.4c4.4 0 8-3.6 8-8.1l-.2-31.7c78.3-6.9 134.3-48.8 134.3-124-.1-69.4-44.2-100.4-109-116.4zm-68.6-16.2c-5.6-1.6-10.3-3.1-15-5-33.8-12.2-49.5-31.9-49.5-57.3 0-36.3 27.5-57 64.5-61.7v124zM534.3 677V543.3c3.1.9 5.9 1.6 8.8 2.2 47.3 14.4 63.2 34.4 63.2 65.1 0 39.1-29.4 62.6-72 66.4z"
        />
      </svg>
    </span>
    <span>
      <a
        href="/deals"
        onClick={[Function]}
      >
        Сделки
      </a>
    </span>
  </li>
  <li
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={
      Object {
        "paddingLeft": 24,
      }
    }
  >
    <span
      aria-label="read"
      className="anticon anticon-read"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="read"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M928 161H699.2c-49.1 0-97.1 14.1-138.4 40.7L512 233l-48.8-31.3A255.2 255.2 0 00324.8 161H96c-17.7 0-32 14.3-32 32v568c0 17.7 14.3 32 32 32h228.8c49.1 0 97.1 14.1 138.4 40.7l44.4 28.6c1.3.8 2.8 1.3 4.3 1.3s3-.4 4.3-1.3l44.4-28.6C602 807.1 650.1 793 699.2 793H928c17.7 0 32-14.3 32-32V193c0-17.7-14.3-32-32-32zM324.8 721H136V233h188.8c35.4 0 69.8 10.1 99.5 29.2l48.8 31.3 6.9 4.5v462c-47.6-25.6-100.8-39-155.2-39zm563.2 0H699.2c-54.4 0-107.6 13.4-155.2 39V298l6.9-4.5 48.8-31.3c29.7-19.1 64.1-29.2 99.5-29.2H888v488zM396.9 361H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm223.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c0-4.1-3.2-7.5-7.1-7.5H627.1c-3.9 0-7.1 3.4-7.1 7.5zM396.9 501H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm416 0H627.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5z"
        />
      </svg>
    </span>
    <span>
      <a
        href="/knowledge"
        onClick={[Function]}
      >
        База знаний
      </a>
    </span>
  </li>
</ul>
`;

exports[`render correct with collapsed true 1`] = `
<ul
  className="ant-menu ant-menu-light ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
  onMouseEnter={[Function]}
  onTransitionEnd={[Function]}
  role="menu"
  style={Object {}}
>
  <li
    className="ant-menu-item ant-menu-item-selected"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
  >
    <span
      aria-label="user"
      className="anticon anticon-user"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="user"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
        />
      </svg>
    </span>
    <span>
      <a
        href="/clients"
        onClick={[Function]}
      >
        Клиенты
      </a>
    </span>
  </li>
  <li
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
  >
    <span
      aria-label="calendar"
      className="anticon anticon-calendar"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="calendar"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
        />
      </svg>
    </span>
    <span>
      <a
        href="/tasks"
        onClick={[Function]}
      >
        Планы
      </a>
    </span>
  </li>
  <li
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
  >
    <span
      aria-label="dollar"
      className="anticon anticon-dollar"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="dollar"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm47.7-395.2l-25.4-5.9V348.6c38 5.2 61.5 29 65.5 58.2.5 4 3.9 6.9 7.9 6.9h44.9c4.7 0 8.4-4.1 8-8.8-6.1-62.3-57.4-102.3-125.9-109.2V263c0-4.4-3.6-8-8-8h-28.1c-4.4 0-8 3.6-8 8v33c-70.8 6.9-126.2 46-126.2 119 0 67.6 49.8 100.2 102.1 112.7l24.7 6.3v142.7c-44.2-5.9-69-29.5-74.1-61.3-.6-3.8-4-6.6-7.9-6.6H363c-4.7 0-8.4 4-8 8.7 4.5 55 46.2 105.6 135.2 112.1V761c0 4.4 3.6 8 8 8h28.4c4.4 0 8-3.6 8-8.1l-.2-31.7c78.3-6.9 134.3-48.8 134.3-124-.1-69.4-44.2-100.4-109-116.4zm-68.6-16.2c-5.6-1.6-10.3-3.1-15-5-33.8-12.2-49.5-31.9-49.5-57.3 0-36.3 27.5-57 64.5-61.7v124zM534.3 677V543.3c3.1.9 5.9 1.6 8.8 2.2 47.3 14.4 63.2 34.4 63.2 65.1 0 39.1-29.4 62.6-72 66.4z"
        />
      </svg>
    </span>
    <span>
      <a
        href="/deals"
        onClick={[Function]}
      >
        Сделки
      </a>
    </span>
  </li>
  <li
    className="ant-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
  >
    <span
      aria-label="read"
      className="anticon anticon-read"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="read"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M928 161H699.2c-49.1 0-97.1 14.1-138.4 40.7L512 233l-48.8-31.3A255.2 255.2 0 00324.8 161H96c-17.7 0-32 14.3-32 32v568c0 17.7 14.3 32 32 32h228.8c49.1 0 97.1 14.1 138.4 40.7l44.4 28.6c1.3.8 2.8 1.3 4.3 1.3s3-.4 4.3-1.3l44.4-28.6C602 807.1 650.1 793 699.2 793H928c17.7 0 32-14.3 32-32V193c0-17.7-14.3-32-32-32zM324.8 721H136V233h188.8c35.4 0 69.8 10.1 99.5 29.2l48.8 31.3 6.9 4.5v462c-47.6-25.6-100.8-39-155.2-39zm563.2 0H699.2c-54.4 0-107.6 13.4-155.2 39V298l6.9-4.5 48.8-31.3c29.7-19.1 64.1-29.2 99.5-29.2H888v488zM396.9 361H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm223.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c0-4.1-3.2-7.5-7.1-7.5H627.1c-3.9 0-7.1 3.4-7.1 7.5zM396.9 501H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm416 0H627.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5z"
        />
      </svg>
    </span>
    <span>
      <a
        href="/knowledge"
        onClick={[Function]}
      >
        База знаний
      </a>
    </span>
  </li>
</ul>
`;
